pandas/_libs/groupby.pyx

```diff
@@ -729,6 +730,10 @@ def group_sum(
     sumx = np.zeros((<object>out).shape, dtype=(<object>out).base.dtype)
     compensation = np.zeros((<object>out).shape, dtype=(<object>out).base.dtype)
 
+    if is_string:
+        # for strings start with empty string instead of 0 as `initial` value
+        sumx = np.full((<object>out).shape, "", dtype=object)
```

Comment:
Would passing “initial” be more general/idiomatic?

Reply:
I was initially thinking that as well, but then because this would in practice only be used for the specific case of strings, I thought to be more explicit about that fact. But in both cases I have to pass it down a few layers, so whether it is then called initial or is_string actually does not matter much, and `initial` at least makes it clearer where it is called from the EA _groupby_ops what the purpose is of passing it down. 
So updated to use `initial` in the last commit